sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -170,7 +211,11 @@ public void verbose(String format, Object... args) {
      */
     public void info(String message) {
         if (logger.isInfoEnabled()) {
-            logger.info(removeNewLinesFromLogMessage(message));
+            if (globalContextSerialized.isEmpty()) {
```

Comment:
nit: Do you think it would be beneficial to introduce a boolean field that represents if this has been set or not (which can be determined in the constructor), mostly to aid in readability (e.g. `if (globalContextIsEmpty) { ... }`), but also slightly to avoid all of the isEmpty calls?